### PR TITLE
Peer and block producer information

### PR DESCRIPTION
**Pull request by Neurosploit**
This pull request adds additonal information about the peer and block producer count.
It also adds the option to selectively change the peers grid page size.
Preview of how this looks:

image

### CODE REVIEW

**Comment by Neurosploit** [comment]
@fenix2222 can you review this when you have time?
Note: only tested on 10k startup, 50k I haven't tested but I don't expect this to break that.

File: Stratis.FederatedSidechains.AdminDashboard.csproj [hunk 1]
```diff
@@ -20,4 +27,422 @@
     <PackageReference Include="RestSharp" Version="106.6.9" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (Neurosploit) [inline Stratis.FederatedSidechains.AdminDashboard.csproj]: I'll remove this clutter, leftovers from trying to move wwwroot to ClientApp folder and let the bundler build to wwwroot, sadly that doesn't move fonts/images so a different packer is needed if you want that, for now I just added build output to the gitignore so it doesn't get added to changes.

File: Views/Shared/Dashboard.cshtml [hunk 2]
```diff
@@ -11,413 +11,427 @@
 }
 <partial name="Modals" />
 <div id="@(ViewBag.DisplayLoader == true ? "loading-content" : "")">
-<script type="text/javascript">
-    $(function() {
-        var settings = { searching: false, pageLength: 5, lengthMenu: false, lengthChange: false, "bDestroy": true };
-        var settings2 = { searching: false, pageLength: 10, lengthMenu: false, lengthChange: false, "bDestroy": true };
+    <script type="text/javascript">
+        $(function () {
+            var peerSettings = { searching: false, pageLength: 5, "bDestroy": true, lengthChange: true, lengthMenu: [[5, 10, 25, 50, -1], [5, 10, 25, 50, "All"]], stateSave: true };
```
Comment (Neurosploit) [inline Views/Shared/Dashboard.cshtml]: Main changes are introducing peerSettings here.

File: Views/Shared/Dashboard.cshtml [hunk 3]
```diff
@@ -11,413 +11,427 @@
...
+                                        <li class="list-inline-item"><strong>Federation Wallet balance :</strong></li>
+                                        <li class="list-inline-item text-green">Confirmed: @Model.StratisNode.ConfirmedBalanceFed.ToString("N8") @Model.StratisNode.CoinTicker</li>
+                                        <li class="list-inline-item text-muted">/</li>
+                                        <li class="list-inline-item text-muted">Unconfirmed: @Model.StratisNode.UnconfirmedBalanceFed.ToString("N8") @Model.StratisNode.CoinTicker</li>
+                                    </ul>
+                                </li>
+                            }
+                            <li class="col-12 col-xl-6">
+                                <ul class="list-inline">
+                                    <li class="list-inline-item"><strong>Total Peers / In / Out:</strong></li>
```
Comment (Neurosploit) [inline Views/Shared/Dashboard.cshtml]: And adding peer info with a new list item here (copy on sidechain node side)

File: Stratis.FederatedSidechains.AdminDashboard.sln [hunk 4]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (ArcSin2000X) [inline Stratis.FederatedSidechains.AdminDashboard.sln]: Please remove the solution file, this is single project

File: Models/SidechainNodeModel.cs [hunk 5]
```diff
@@ -10,5 +10,9 @@ public class SidechainNodeModel : StratisNodeModel
         public string BlockProducerHits { get; set; }
 
         public decimal BlockProducerHitsValue { get; set; }
+
+        public int BlockProducersMining { get; set; }
```
Comment (ArcSin2000X) [inline Models/SidechainNodeModel.cs]: How are these different to what is already in fields related to block producer hits

Comment (Neurosploit) [reply Models/SidechainNodeModel.cs]: It contains (...%) in the string as well, I'll split it off and use that instead in the model and then remove these fields.

File: Stratis.FederatedSidechains.AdminDashboard.sln (hunk 4 above)
Comment (Neurosploit) [reply Stratis.FederatedSidechains.AdminDashboard.sln]: Ok

**Pull request merged**